### ISSUE TITLE
Error encountered when visiting Debian packages

### ISSUE THREAD

**Issue by JonoYang** [nexB/purldb#34]
```
purldb-visitor-1  | ERROR:minecode.management.commands.run_visit:Visit error for URI: http://mirrors.aliyun.com/ubuntu/ls-lR.gz
purldb-visitor-1  | TypeError TypeError('cannot use a string pattern on a bytes-like object')
purldb-visitor-1  | Traceback (most recent call last):
purldb-visitor-1  |   File "/app/minecode/management/commands/run_visit.py", line 297, in visit_uri
purldb-visitor-1  |     for vuri_count, vuri in enumerate(new_uris_to_visit):
purldb-visitor-1  |   File "/app/minecode/visitors/debian.py", line 94, in get_uris
purldb-visitor-1  |     for entry in ls.parse_directory_listing(content):
purldb-visitor-1  |   File "/app/minecode/ls.py", line 167, in parse_directory_listing
purldb-visitor-1  |     if parser.ignores_line(line):
purldb-visitor-1  |   File "/usr/local/lib/python3.9/site-packages/ftputil/stat.py", line 120, in ignores_line
purldb-visitor-1  |     match = self._total_regex.search(line)
purldb-visitor-1  | TypeError: cannot use a string pattern on a bytes-like object
purldb-visitor-1  | 
purldb-visitor-1  | INFO:minecode.management.commands.run_visit:Visiting http://ftp.debian.org/debian/ls-lR.gz
purldb-visitor-1  | ERROR:minecode.management.commands.run_visit:Visit error for URI: http://ftp.debian.org/debian/ls-lR.gz
purldb-visitor-1  | TypeError TypeError('cannot use a string pattern on a bytes-like object')
purldb-visitor-1  | Traceback (most recent call last):
purldb-visitor-1  |   File "/app/minecode/management/commands/run_visit.py", line 297, in visit_uri
purldb-visitor-1  |     for vuri_count, vuri in enumerate(new_uris_to_visit):
purldb-visitor-1  |   File "/app/minecode/visitors/debian.py", line 94, in get_uris
purldb-visitor-1  |     for entry in ls.parse_directory_listing(content):
purldb-visitor-1  |   File "/app/minecode/ls.py", line 167, in parse_directory_listing
purldb-visitor-1  |     if parser.ignores_line(line):
purldb-visitor-1  |   File "/usr/local/lib/python3.9/site-packages/ftputil/stat.py", line 120, in ignores_line
purldb-visitor-1  |     match = self._total_regex.search(line)
purldb-visitor-1  | TypeError: cannot use a string pattern on a bytes-like object
```

**Comment by 35C4n0r** [comment]
@JonoYang this has been covered in the #23.
> https://docs.python.org/3/library/gzip.html#gzip.open quotes that The mode argument can be any of 'r', 'rb', 'a', 'ab', 'w', 'wb', 'x' or 'xb' for binary mode, or 'rt', 'at', 'wt', or 'xt' for text mode. The default is 'rb', since we later need do operation on this content ( see https://github.com/nexB/purldb/blob/main/minecode/visitors/debian.py#L89 ) we need to either use 'rt' or decode("utf-8").

**Comment by JonoYang** [comment]
@35C4n0r Thanks! I'll try out the fix locally

**Comment by JonoYang** [comment]
I've applied the patch from #23 and I now get these errors when running the debian visitor:
```
ValueError ValueError('Unknown Debian archive filename format: 1oom_1.0-2.dsc')
Traceback (most recent call last):
  File "/home/jono/nexb/src/purldb/minecode/management/commands/run_visit.py", line 297, in visit_uri
    for vuri_count, vuri in enumerate(new_uris_to_visit):
  File "/home/jono/nexb/src/purldb/minecode/visitors/debian.py", line 112, in get_uris
    name, version, arch = debian_inspector.package.get_nva(file_name)
  File "/home/jono/nexb/src/purldb/venv/lib/python3.10/site-packages/debian_inspector/package.py", line 133, in get_nva
    raise ValueError(
ValueError: Unknown Debian archive filename format: 1oom_1.0-2.dsc

INFO:minecode.management.commands.run_visit:Visiting http://archive.ubuntu.com/ubuntu/ls-lR.gz
ERROR:minecode.management.commands.run_visit:Visit error for URI: http://archive.ubuntu.com/ubuntu/ls-lR.gz
ValueError ValueError('Unknown Debian archive filename format: Contents-amd64.gz')
Traceback (most recent call last):
  File "/home/jono/nexb/src/purldb/minecode/management/commands/run_visit.py", line 297, in visit_uri
    for vuri_count, vuri in enumerate(new_uris_to_visit):
  File "/home/jono/nexb/src/purldb/minecode/visitors/debian.py", line 112, in get_uris
    name, version, arch = debian_inspector.package.get_nva(file_name)
  File "/home/jono/nexb/src/purldb/venv/lib/python3.10/site-packages/debian_inspector/package.py", line 133, in get_nva
    raise ValueError(
ValueError: Unknown Debian archive filename format: Contents-amd64.gz
```
I think the first error is related to https://github.com/nexB/purldb/pull/30

**Comment by 35C4n0r** [comment]
> I've applied the patch from #23 and I now get these errors when running the debian visitor:
> 
> ```
> ValueError ValueError('Unknown Debian archive filename format: 1oom_1.0-2.dsc')
> Traceback (most recent call last):
>   File "/home/jono/nexb/src/purldb/minecode/management/commands/run_visit.py", line 297, in visit_uri
>     for vuri_count, vuri in enumerate(new_uris_to_visit):
>   File "/home/jono/nexb/src/purldb/minecode/visitors/debian.py", line 112, in get_uris
>     name, version, arch = debian_inspector.package.get_nva(file_name)
>   File "/home/jono/nexb/src/purldb/venv/lib/python3.10/site-packages/debian_inspector/package.py", line 133, in get_nva
>     raise ValueError(
> ValueError: Unknown Debian archive filename format: 1oom_1.0-2.dsc
> 
> INFO:minecode.management.commands.run_visit:Visiting http://archive.ubuntu.com/ubuntu/ls-lR.gz
> ERROR:minecode.management.commands.run_visit:Visit error for URI: http://archive.ubuntu.com/ubuntu/ls-lR.gz
> ValueError ValueError('Unknown Debian archive filename format: Contents-amd64.gz')
> Traceback (most recent call last):
>   File "/home/jono/nexb/src/purldb/minecode/management/commands/run_visit.py", line 297, in visit_uri
>     for vuri_count, vuri in enumerate(new_uris_to_visit):
>   File "/home/jono/nexb/src/purldb/minecode/visitors/debian.py", line 112, in get_uris
>     name, version, arch = debian_inspector.package.get_nva(file_name)
>   File "/home/jono/nexb/src/purldb/venv/lib/python3.10/site-packages/debian_inspector/package.py", line 133, in get_nva
>     raise ValueError(
> ValueError: Unknown Debian archive filename format: Contents-amd64.gz
> ```
> 
> I think the first error is related to #30

@JonoYang PR #30 was closed, new PR #41, that PR will solve this issue, please review.
